src/test/java/org/folio/search/controller/IndexManagementIT.java

```diff
@@ -114,6 +127,41 @@ void runReindex_positive_locations() throws Exception {
     });
   }
 
+  @ParameterizedTest
+  @EnumSource(value = ResourceNameEnum.class, names = {"LINKED_DATA_WORK", "LINKED_DATA_AUTHORITY"})
+  void runReindex_shouldRecreate_linkedDataResourcesIndexes(ResourceNameEnum resourceNameEnum) throws Exception {
+    var resourceName = resourceNameEnum.getValue();
+    var request = post(ApiEndpoints.reindexPath())
+      .content(asJsonString(new ReindexRequest().resourceName(resourceNameEnum).recreateIndex(true)))
+      .headers(defaultHeaders())
+      .header(XOkapiHeaders.URL, okapi.getOkapiUrl())
+      .contentType(MediaType.APPLICATION_JSON);
+    var indexIdBeforeReindex = getIndexId(resourceName);
+
+    mockMvc.perform(request)
+      .andExpect(status().isOk());
+
+    await().atMost(FIVE_SECONDS)
+      .pollInterval(ONE_HUNDRED_MILLISECONDS)
+      .untilAsserted(() -> {
+        assertNotEquals(indexIdBeforeReindex, getIndexId(resourceName));
+        assertThat(countDefaultIndexDocument(resourceName)).isEqualTo(0);
+      });
+  }
+
+  @ParameterizedTest
+  @MethodSource("requestBuilderDataProvider")
+  void runReindex_shouldNotRecreate_linkedDataResourcesIndexes(RequestBuilder requestBuilder) throws Exception {
+    var indexIdBeforeReindex = getIndexId(LINKED_DATA_WORK_RESOURCE);
+
+    mockMvc.perform(requestBuilder)
+      .andExpect(status().isOk());
+
+    await().atMost(FIVE_SECONDS)
+      .pollInterval(ONE_HUNDRED_MILLISECONDS)
+      .untilAsserted(() -> assertEquals(indexIdBeforeReindex, getIndexId(LINKED_DATA_WORK_RESOURCE)));
```

Comment:
What do you asserting there?

Reply:
I'm asserting that the initial index hasn't been recreated.